.github/workflows/build-test.yml

```diff
@@ -19,43 +19,108 @@ jobs:
+      examples: ${{ steps.filter.outputs.changes }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            java-chatroom: play-java-chatroom-example/**
+            java-compile-di: play-java-compile-di-example/**
+            java-dagger2: play-java-dagger2-example/**
+            java-ebean: play-java-ebean-example/**
+            java-fileupload: play-java-fileupload-example/**
+            java-forms: play-java-forms-example/**
+            java-grpc: play-java-grpc-example/**
+            java-hello-world: play-java-hello-world-tutorial/**
+            java-jpa: play-java-jpa-example/**
+            java-pekko-cluster: play-java-pekko-cluster-example/**
+            java-rest-api: play-java-rest-api-example/**
+            java-starter: play-java-starter-example/**
+            java-streaming: play-java-streaming-example/**
+            java-websocket: play-java-websocket-example/**
+            scala-anorm: play-scala-anorm-example/**
+            scala-chatroom: play-scala-chatroom-example/**
+            scala-compile-di: play-scala-compile-di-example/**
+            scala-fileupload: play-scala-fileupload-example/**
+            scala-forms: play-scala-forms-example/**
+            scala-grpc: play-scala-grpc-example/**
+            scala-hello-world: play-scala-hello-world-tutorial/**
+            scala-isolated-slick: play-scala-isolated-slick-example/**
+            scala-log4j2: play-scala-log4j2-example/**
+            scala-macwire-di: play-scala-macwire-di-example/**
+            scala-rest-api: play-scala-rest-api-example/**
+            scala-secure-session: play-scala-secure-session-example/**
+            scala-slick: play-scala-slick-example/**
+            scala-starter: play-scala-starter-example/**
+            scala-streaming: play-scala-streaming-example/**
+            scala-tls: play-scala-tls-example/**
+            scala-websocket: play-scala-websocket-example/**
 
-  gradle-tests:
-    name: Gradle Tests
-    uses: playframework/.github/.github/workflows/cmd.yml@v3
+  java-chatroom:
+    name: "Java :: Chatroom"
+    uses: playframework/.github/.github/workflows/cmd.yml@v4
+    needs: changes
+    if: ${{ (github.event_name != 'pull_request' && !failure() && !cancelled()) || contains(fromJSON(needs.changes.outputs.examples), 'java-chatroom')}}
     with:
-      java: 17, 11
+      java: 17, 21
       scala: 2.13.x, 3.x
+      gradle-build-root: play-java-chatroom-example
+      add-dimensions: >-
+        { "server": [ "pekko", "netty"] }
       cmd: |
-        ./test-gradle.sh
+        cd play-java-chatroom-example
+        if [[ $MATRIX_SERVER == netty ]]; then
+          find -name build.sbt | xargs sed -i 's/\/\/.enablePlugins(PlayNettyServer)/.enablePlugins(PlayNettyServer)/g'
+        fi
+        sbt ++$MATRIX_SCALA test
+        ./gradlew -Dscala.version="$MATRIX_SCALA" check
```

Comment:
so this needs to be duplicated 31 times?

Reply:
To me it isn't duplicate, it's a moving from scripts folder in every examples, that I want to delete in the future 😂 But technically, you're absolutely right.

Reply:
But this introduced much more duplication, we have
```sh
find -name build.sbt | xargs sed -i 's/\/\/.enablePlugins(PlayNettyServer)/.enablePlugins(PlayNettyServer)/g'
```
just once in the test.sh file now, but now you have to add that (and the if condition) 31 times...